gatsby-node.js

```diff
@@ -152,19 +152,11 @@ exports.sourceNodes = async (
     };
 
     createNode(node);
-    return processedData.childNodes;
-  });
-
-  let flattenedChildNodes = await Promise.all(childNodes).then(nodes =>
-    nodes.reduce(
-      (accumulator, currentValue) => accumulator.concat(currentValue),
-      []
-    )
-  );
 
-  return Promise.all(flattenedChildNodes).then(nodes => {
-    nodes.forEach(node => createNode(node));
-  });
+    await Promise.all(processedData.childNodes).then(nodes => {
+      nodes.forEach(node => createNode(node));
+    });
+  }, { concurrency: 100 });
```

Comment:
```suggestion
  }, { concurrency: 201 });
```

Just jumping on this since I happened to see your Travis build running while I was taking a look at mine! I had also been following your issue since I use Airtable in a similar fashion outside of my gatsby project. Since their official documentation states you will be rate limited if you exceed 5 requests per second, what do you think of my suggestion?

Reply:
The concurrency option is a limit on the maximum number of outstanding requests allowed rather than a time limit between requests.

The 5 request per second limit is for API requests whereas these requests are against the Airtable file attachment CDN. I think the problem was that without any limiting the plugin was running up against DDoS protection for larger numbers of file attachments, not their API rate limiting.

So the goal was to chunk requests out just enough not to trigger the blocks. 100 was an arbitrary value that worked. An edit in the spirit of your suggestion would be to use the value 5 (for 5 active requests at a time) rather than 201, but my understanding of the situation suggests that it’s not necessary.

Reply:
Aha, thanks for the enlightening comment. That makes sense!